snuba/query/project_extension.py

```diff
@@ -52,15 +81,12 @@ def process_query(
             self,
             query: Query,
             extension_data: ExtensionData,
-            request_settings: RequestSettings,
+            request_query_settings: RequestQuerySettings,
     ) -> None:
-        project_ids = util.to_list(extension_data['project'])
-
-        if project_ids:
-            query.add_conditions([('project_id', 'IN', project_ids)])
+        super().process_query(query, extension_data, request_query_settings)
 
-        if not request_settings.turbo:
-            final, exclude_group_ids = get_projects_query_flags(project_ids)
+        if not request_query_settings.get_turbo():
+            final, exclude_group_ids = get_projects_query_flags(self._project_ids)
```

Comment:
I think this uncovers a smell. We are depending on super to leave the state of the class in a state you expect. I would remove the process_query method from this class, leave it only in the parent class and let the parent class delegate to a protected method everything from line 88 to line 99 like in a "template method" pattern